integration-tests/cypress/commands/api/device.ts

```diff
@@ -0,0 +1,539 @@
+		  expect(users[index].email).to.equal(expectedUsers[index].email);
+	  }
+       };
+  });
+});
+
+Cypress.Commands.add(
+  "apiAddUserToDevice",
+  (deviceAdminUser: string, userName: string, device: string, admin: boolean = false, statusCode: number = 200) => {
+    logTestDescription(
+      `${deviceAdminUser} Adding user '${userName}' to device '${device}'`,
+      { user: userName, device }
+    );
+
+    makeAuthorizedRequestWithStatus(
+      {
+        method: "POST",
+        url: v1ApiPath("devices/users"),
+        body: {
+          deviceId: getCreds(device).id,
+          admin: admin,
+          username: getTestName(userName)
+        }
+      },
+      deviceAdminUser,
+      statusCode
+    );
+  }
+);
+
+Cypress.Commands.add(
+  "apiRemoveUserFromDevice",
+  (deviceAdminUser: string, userName: string, device: string, statusCode: number = 200) => {
+    logTestDescription(
+      `${deviceAdminUser} Removing user '${userName}' to device '${device}'`,
+      { user: userName, device }
+    );
+
+    makeAuthorizedRequestWithStatus(
+      {
+        method: "DELETE",
+        url: v1ApiPath("devices/users"),
+        body: {
+          deviceId: getCreds(device).id,
+          admin: "false",
+          username: getTestName(userName)
+        }
+      },
+      deviceAdminUser,
+      statusCode
+    );
+  }
+);
+
+
+
+
+Cypress.Commands.add("apiUploadRecording", (cameraName, id) => {
+  const recordingsUrl = v1ApiPath('recordings');
+  const deviceName = getTestName(cameraName);
+
+  const data = {
+       type: 'thermalRaw'
+  };
+
+  const fileName="bird_"+id+".cptv";
+  const uniqueName=deviceName+"_"+id;
+  const aliasName=deviceName+"_a"+id;
+
+
+  const creds = getCreds(cameraName);
+  uploadFileRequest(fileName, uniqueName, aliasName, recordingsUrl, data, creds)
+
+});
+
+
+Cypress.Commands.add("apiUploadEvent", (cameraName) => {
+  const eventsUrl = v1ApiPath('events');
+  const deviceName = getTestName(cameraName);
+  const creds = getCreds(cameraName);
+
+  const data = {
+       description: {
+	   type: 'throttle'
+       },
+       dateTimes: ['2017-11-13T00:47:51.160Z']
+  };
+
+  makeAuthorizedRequest(
+      {
+        method: "POST",
+        url: eventsUrl,
+        body:  data 
+      },
+     cameraName
+    );
+
+});
+
+Cypress.Commands.add("apiCheckDeviceHasRecordings", (username, deviceName,count) => {
+  const user = getCreds(username);
+  const camera = getCreds(deviceName);
+  const fullUrl = v1ApiPath('')+encodeURI('recordings?where={"DeviceId":'+camera.id+'}');
+
+  cy.request({
+    url: fullUrl,
+    headers: user.headers
+  }).then((request) => {
+    expect(request.body.count).to.equal(count);
+  });
+});
+
+
+Cypress.Commands.add("apiCheckEventUploaded", (username, deviceName, eventType) => {
+  const user = getCreds(username);
+  const camera = getCreds(deviceName);
+  const eventURL = v1ApiPath('events')+'?deviceId='+camera.id;
```

Comment:
dsimilar to above use the optional query params option for v1ApiPath()

Reply:
Agree